proposals/static-delegates.md

```diff
@@ -0,0 +1,74 @@
+# Static Delegates
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a general-purpose, lightweight callback capability to the C# language.
+
+## Motivation
+[motivation]: #motivation
+
+Today, users have the ability to create callbacks using the `System.Delegate` type. However, these are fairly heavyweight (such as requiring a heap allocation and always having handling for chaining callbacks together).
+
+Additionally, `System.Delegate` does not provide the best interop with unmanaged function pointers, namely due being non-blittable and requiring marshalling anytime it crosses the managed/unmanaged boundary.
+
+With a few minor tweaks, we could provide a new type of delegate that is lightweight, general-purpose, and interops well with native code.
+
+## Detailed design
+[design]: #detailed-design
+
+One would declare a static delegate via the following:
+
+```C#
+static delegate int Func()
```

Comment:
I prefer `static delegate` over `value delegate`.  

The `Value` pattern is used today for types which are being made a `struct` type.  The `Value` prefix is given to distinguish them from the well know `class` types.  If there were a well known base type for this feature then I agree that calling it `ValueDelegate` would make sense.  That's not the case though here, there is in fact no common base type to refer to.  

In this case it's a delegate that can only refer to `static` members.  There are existing examples in the language of using `static` to modify existing types such that it can only contain `static` content: `static class`.  The `static delegate` notation is just feeding into this existing usage.